core/types/transaction.go

```diff
@@ -622,13 +623,42 @@ func (tx *Transaction) EffectiveGasTipIntCmp(other *big.Int, baseFee *big.Int) i
 }
 
 // Hash returns the transaction hash.
-func (tx *Transaction) Hash() (h common.Hash) {
+func (tx *Transaction) Hash(location ...byte) (h common.Hash) {
```

Comment:
You should not need to specify location. Everything about origin and destination locations can be fully determined by the transaction type and data.

Reply:
ok, I see. We can allow it for now, but please leave an issue to use an atomic to cache the value after the first ecrecover.